server/src/main/java/org/apache/cassandra/sidecar/coordination/SidecarHttpHealthProvider.java

```diff
@@ -0,0 +1,81 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.coordination;
+
+import java.util.concurrent.CompletableFuture;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.vertx.core.Future;
+import org.apache.cassandra.sidecar.client.SidecarClient;
+import org.apache.cassandra.sidecar.client.retry.BasicRetryPolicy;
+import org.apache.cassandra.sidecar.common.client.SidecarInstance;
+import org.apache.cassandra.sidecar.common.client.SidecarInstanceImpl;
+import org.apache.cassandra.sidecar.common.response.HealthResponse;
+import org.apache.cassandra.sidecar.config.SidecarConfiguration;
+import org.apache.cassandra.sidecar.config.SidecarPeerHealthConfiguration;
+import org.apache.cassandra.sidecar.utils.SidecarClientProvider;
+
+
+/**
+ * Provides the health of a Sidecar instance over HTTP API, retrying to
+ * confirm Sidecar is DOWN for extended period of time.
+ */
+@Singleton
+public class SidecarHttpHealthProvider implements SidecarPeerHealthProvider
+{
+    private final SidecarPeerHealthConfiguration config;
+    private final SidecarClientProvider clientProvider;
+
+    @Inject
+    public SidecarHttpHealthProvider(SidecarConfiguration sidecarConfiguration, SidecarClientProvider clientProvider)
+    {
+        this.config = sidecarConfiguration.sidecarPeerHealthConfiguration();
+        this.clientProvider = clientProvider;
+    }
+
+    @Override
+    public Future<Health> health(SidecarInstance instance)
+    {
+        try
+        {
+            SidecarClient client = clientProvider.get();
+            CompletableFuture<HealthResponse> healthRequest = client.executeRequestAsync(client.requestBuilder()
+                                                                                               .singleInstanceSelectionPolicy(
+                                                                                               new SidecarInstanceImpl(instance.hostname(),
+                                                                                                                       instance.port()))
+                                                                                               .retryPolicy(retryPolicy())
+                                                                                               .sidecarHealthRequest()
+                                                                                               .build());
+            return Future.fromCompletionStage(healthRequest)
+                         .map(healthResponse -> healthResponse.isOk()
+                                                ? Health.OK
+                                                : Health.DOWN);
+        }
+        catch (Exception e)
+        {
+            return Future.succeededFuture(Health.DOWN);
```

Comment:
Looks like that `UNKNOWN` is not really used anywhere. Can we remove it? If it make sense to have `UNKNOWN`, can you use it?